Source/Core/VideoCommon/NetPlayChatUI.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <deque>
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+
+struct ImVec4;
+
+class NetPlayChatUI
+{
+public:
+  explicit NetPlayChatUI(std::function<void(const std::string&)> callback);
```

Comment:
destructor should also be defaulted in the cpp file, since the class contains non-trivial types.